ui/pages/Search/components/filters/PaMoiSelector.jsx

```diff
@@ -0,0 +1,79 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import { Multiselect } from 'shared/components/form/Inputs'
+import { semanticShouldUpdate } from 'shared/utils/semanticUtils'
+import { moiToMoiInitials, panelAppLocusListReducer } from 'shared/utils/panelAppUtils'
+import { PANEL_APP_MOI_OPTIONS } from 'shared/utils/constants'
+
+class PaMoiDropdown extends React.Component {
+
+  static propTypes = {
+    locus: PropTypes.object,
+    onChange: PropTypes.func,
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    return semanticShouldUpdate(this, nextProps, nextState)
+  }
+
+  handleMOIselect = (selectedMOIs) => {
+    const { locus, onChange } = this.props
+    const { locusList } = locus
+
+    const panelAppItems = locusList.items?.filter((item) => {
+      let result = true
+      if (selectedMOIs && selectedMOIs.length !== 0) {
+        const initials = moiToMoiInitials(item.pagene?.modeOfInheritance)
+        if (initials.length === 0) initials.push('other')
+        result = selectedMOIs.filter(moi => initials.includes(moi)).length !== 0
+      }
+      return result
+    }).reduce(panelAppLocusListReducer, {})
+
+    onChange({ ...panelAppItems })
+  }
+
+  moiOptions = () => {
+    const { locus } = this.props
+    const { locusList } = locus
+
+    const initials = locusList.items?.reduce((acc, gene) => {
+      moiToMoiInitials(gene.pagene?.modeOfInheritance).forEach((initial) => {
+        acc[initial] = true
+      })
+      if (moiToMoiInitials(gene.pagene?.modeOfInheritance).length === 0) {
+        acc.other = true
+      }
+      return acc
+    }, {}) || {}
+
+    return PANEL_APP_MOI_OPTIONS.map(moi => ({
```

Comment:
This is an enormous amount of code complexity to add what is honestly not a super important piece of functionality, it seems like you've needed to change the memoization and update behavior of like 5 different components to get this to work. I would rather have a simple, maintainable list of options, as I don't think selecting an option and having no genes show up is a bad user experience. That said, if you feel really strongly about disabling options we can leave this

Reply:
Thank you, I think this feature will be very beneficial to the users, I believe it is worth taking on this extra complexity in the code to free up mental load when curating variants.

This gif demonstrates how it looks when flipping between gene lists:
![seqr-2770](https://user-images.githubusercontent.com/3320713/178984741-105ef7d7-43cf-4c12-ab45-0af63ad1cb83.gif)